nncf/openvino/quantization/weights_compression.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Type, Union
+
+import numpy as np
+import openvino.runtime as ov
+from openvino.runtime import opset9 as opset
+
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVEmbeddingMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVMatMulMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import get_operation_const_op
+from nncf.openvino.graph.nncf_graph_builder import GraphConverter
+from nncf.openvino.graph.node_utils import get_const_value
+from nncf.openvino.graph.node_utils import get_matmul_channel_axes
+
+
+def insert_pre_compression_operations(model: ov.Model, bits: int = 8) -> None:
+    """
+    Compress weights of Linear and Embedding layers to uint8.
+    The result of compression is the same as asymmetric weight quantization.
+
+    :param model: The model to be transformed.
+    :param bits: Number of bits for quantization.
+    """
+    allowed_metatypes_to_const_port = {OVEmbeddingMetatype: [0], OVMatMulMetatype: [0, 1]}
+    level_low = 0
+    level_high = 2**bits - 1
+
+    for node in model.get_ops():
+        # pylint:disable=protected-access
+        metatype = GraphConverter._get_node_metatype(node)
```

Comment:
I would suggest to avoid `# pylint:disable=protected-access` by renaming this function to `get_node_metatype()`. Or maybe the better approach is to have this function inside `nncf/nncf/openvino/graph/metatypes/openvino_metatypes.py`